### PR TITLE
  Implementation of gpfs daemon network to ansible toolkit

### PR DESCRIPTION
**Pull request by Christop1964**
Implementation of gpfs daemon network to ansible toolkit

  Signed-off-by: Christoph Keil chkeil@de.ibm.com

```
[cluster01]
scale-11 scale_cluster_quorum=true scale_cluster_manager=true scale_zimon_collector=true scale_cluster_gui=false is_protocol_node=true  scale_admin_nodename=scale-13  scale_daermon_nodename=scale-13d
scale-12 scale_cluster_quorum=true scale_cluster_manager=true scale_zimon_collector=true scale_cluster_gui=false is_protocol_node=true  scale_admin_nodename=scale-13  scale_daemon_nodename=scale-13d

```
I will create a separate pull request to update the readme.

### CODE REVIEW

**Review by rajan-mis — APPROVED** [review]
Looks good to me.  I also executed Jenkins over this Pull request .  I discussed with @Christop1964 to create pull request for README update. He will be creating separate PR for README update.

File: roles/core/cluster/defaults/main.yml [hunk 1]
```diff
@@ -25,6 +25,9 @@
 ## Spectrum Scale daemon nodename (defaults to node's hostname)
 scale_daemon_nodename: "{{ ansible_hostname }}"
 
+## Spectrum Scale admin nodename (defaults to node's hostname)
+scale_admin_nodename: "{{ ansible_hostname }}"
```
Comment (acch) [inline roles/core/cluster/defaults/main.yml:29]: You might want to set `scale_admin_nodename` to `scale_daemon_nodename` by default. If users change daemon network then admin network will follow these changes automatically...
```
scale_admin_nodename: "{{ scale_daemon_nodename }}"
```

File: roles/core/precheck/tasks/prepare.yml [hunk 2]
```diff
@@ -32,6 +32,17 @@
         line: PubkeyAuthentication yes
         state: present
       notify: reload-sshd
+
+    # prepare | Update ListenAddress at sshd_config for daemon network
+    - name: Update ListenAddress at /etc/ssh/sshd_config
```
Comment (acch) [inline roles/core/precheck/tasks/prepare.yml:37]: I'd suggest to follow this convention when naming tasks:
```
- name: prepare | What does the task do?
```

- In this example, `prepare` is the name of the file, as well as the name of the associated tag.
- `What does the task do?` is a high-level description of what this task enables. It's not so relevant which particular option in which particular file is being modified... but what's the impact of this change? i.e. what does the `ListenAddress` option control and why do we want to set a certain value? Examples: "Enable SSH root login" or "Enable SSH pubkey authentication"...

File: roles/core/precheck/defaults/main.yml [hunk 3]
```diff
@@ -49,6 +49,10 @@ scale_prepare_disable_selinux: false
 ## rules prior to running this role (e.g. as pre_tasks)
 scale_prepare_disable_firewall: false
 
+## For the Spectrum Scale daemon network the 'ListenAddress' needs to be set
+## at /etc/ssh/sshd_config to allow only the admin IP to login.
+## Set the scale_sshd_config_update to 'true' to update the sshd_config
+scale_sshd_config_update: false
```
Comment (acch) [inline roles/core/precheck/defaults/main.yml:55]: With the current code one needs to enable `scale_prepare_enable_ssh_login` AND `scale_sshd_config_update` to enable the task - is this on purpose? If this is on purpose then we should document this dependency.

Also, the name `scale_sshd_config_update` is quite generic - in fact this variable only controls the `ListenAddress` task - should this be reflected in it's name?

**Review by acch — COMMENTED** [review]
Thank Christoph, a few more comments below...

**Pull request merged**